### PR TITLE
SCRUM-164: Add Run Functionality to Programming Questions

### PR DESCRIPTION
**Pull request by daniellandsman1**
This PR addresses [SCRUM-164: Add Run Functionality to /submitCode](https://knightwise.atlassian.net/browse/SCRUM-164).

- This PR modifies the programming question backend to implement a new "Test Run" feature:
  - This new feature allows users to execute their code against the first test case associated with a question.
  - The test run is not stored in the Response table like actual submissions. Instead, they are saved to a new TestRun database table, which just records the IDs of the user and question (foreign keys) and a timestamp.
    - These saved test run entries are used to enforce a maximum limit of test runs per problem per day, currently set to 3. The test run limit and global actual submission limit (currently 10) **ARE ISOLATED AND DO NOT AFFECT EACH OTHER**.
- The `/submitCode` endpoint now takes and returns a new parameter, `isTestRun`. If true, user code is executed and saved as a test run. If false, user code is executed, submitted, graded, and saved as a response.
- Additionally, the code controller logic was tweaked to count compilation/runtime errors toward actual submission/test run attempts, saving them to the database. Unit tests were added to verify this tweak.

- Unit tests were added to verify new functionality:
  - Existing tests in `code.test.js` were modified to use the new `isTestRun` parameter.
  - Further tests were added to verify intended test run behavior, new submission limits, and other edge cases.

- This PR includes the necessary database schema changes to support the new functionality:
  - New **TestRun** table with **USERID** (foreign key), **QUESTION_ID** (foreign key), and **DATETIME**.
  - KW_Testing was synced to these schema changes.

- This PR includes changes to the `swagger.yaml` documentation corresponding to the new changes.

- The changes were tested locally.

- Below: Proof of new test cases passing:
image
image

- Below: Example of new `isTestRun` parameter for an actual code submission. Looks like I exceeded 10 global submissions!
image

- Below: Example of new `isTestRun` parameter for a test run. Note how this doesn't return grading info, just the stdout result of running against the first test case.
image

- Below: Proof of distinct test run limit. Looks like we hit 3 test runs for this question!
image

- Below: But we can still do 3 test runs for another question. The test run limit is per question, not global.
image

Issues closed:
- https://knightwise.atlassian.net/browse/SCRUM-164

### CODE REVIEW

File: backend/controllers/codeController.js [hunk 1]
```diff
@@ -102,30 +105,11 @@ const submitCode = asyncHandler(async (req, res) => {
   // Code can only be submitted by account owner
   const userId = req.user.id; // Set by authMiddleware.js
 
-  // Check if user has exceeded max daily programming question submissions
```
Comment (daniellandsman1) [inline backend/controllers/codeController.js:105]: I didn't delete this part, it's just further down in the file now!

**Review by minervatbd — APPROVED** [review]

**Review by dahlia-factory — APPROVED** [review]
